jobs/build-charms/builder_launchpad.py

```diff
@@ -129,7 +129,11 @@ def _lp_build_log(self, build: Resource) -> str:
 
     def _lp_request_builds(self) -> List[Resource]:
         """Request a charm build for this charm."""
-        req = self._lp_recipe.requestBuilds(channels=self.SNAP_CHANNEL)
+        channels = self.SNAP_CHANNEL
+        charmcraft_channel_file = Path(self.src_path) / ".charmcraft-channel"
+        if charmcraft_channel_file.exists():
+            channels["charmcraft"] = charmcraft_channel_file.read_text().strip()
+        req = self._lp_recipe.requestBuilds(channels=channels)
```

Comment:
let the charm describe which version of charmcraft it wishes to use to build within the repo/branch `.charmcraft-channel`